### ISSUE TITLE
Failed: NewConnectionPoolSpec.The host-level client infrastructure should never close hot connections when minConnections key is given and >0 (minConnections = 5)

### ISSUE THREAD

**Issue by 2m** [akka/akka-http#1798]
https://jenkins.akka.io:8498/job/akka-http-nightly-release-10.0-against-akka-master/34/testReport/junit/akka.http.impl.engine.client/NewConnectionPoolSpec/The_host_level_client_infrastructure_should_never_close_hot_connections_when_minConnections_key_is_given_and__0__minConnections___5_/

```
java.lang.AssertionError: assertion failed: timeout (5 seconds) during expectMsgClass waiting for class akka.stream.testkit.TestSubscriber$OnNext
	at scala.Predef$.assert(Predef.scala:170)
	at akka.testkit.TestKitBase$class.expectMsgClass_internal(TestKit.scala:508)
	at akka.testkit.TestKitBase$class.expectMsgType(TestKit.scala:481)
	at akka.testkit.TestKit.expectMsgType(TestKit.scala:890)
	at akka.stream.testkit.TestSubscriber$ManualProbe.expectNextN(StreamTestKit.scala:374)
	at akka.http.impl.engine.client.ConnectionPoolSpec$$anonfun$1$$anonfun$apply$mcV$sp$14$$anon$17$$anonfun$30.apply(ConnectionPoolSpec.scala:344)
	at akka.http.impl.engine.client.ConnectionPoolSpec$$anonfun$1$$anonfun$apply$mcV$sp$14$$anon$17$$anonfun$30.apply(ConnectionPoolSpec.scala:339)
	at scala.collection.immutable.Range.foreach(Range.scala:160)
	at akka.http.impl.engine.client.ConnectionPoolSpec$$anonfun$1$$anonfun$apply$mcV$sp$14$$anon$17.<init>(ConnectionPoolSpec.scala:339)
	at akka.http.impl.engine.client.ConnectionPoolSpec$$anonfun$1$$anonfun$apply$mcV$sp$14.apply(ConnectionPoolSpec.scala:329)
	at akka.http.impl.engine.client.ConnectionPoolSpec$$anonfun$1$$anonfun$apply$mcV$sp$14.apply(ConnectionPoolSpec.scala:329)
	at org.scalatest.OutcomeOf$class.outcomeOf(OutcomeOf.scala:85)
```

Looks like another case of #1726

**Comment by chbatey** [comment]
https://jenkins.akka.io:8498/job/akka-http-nightly-release-10.0-against-akka-master/36/testReport/junit/akka.http.impl.engine.client/NewConnectionPoolSpec/The_host_level_client_infrastructure_should_never_close_hot_connections_when_minConnections_key_is_given_and__0__minConnections___5_/

**Comment by jrudolph** [comment]
Probably fixed by #1776 once all backports are merged (#1824).
